lib/cloudwatch-integration.js

```diff
@@ -72,16 +72,24 @@ lib.upload = function(aws, groupName, streamName, logEvents, retentionInDays, cb
 
       lib._postingEvents = true;
       debug('send to aws');
-      aws.putLogEvents(payload, function(err) {
+      aws.putLogEvents(payload, function(err, data) {
         if (err) {
-          if (err.code === 'InvalidSequenceTokenException') {
-            debug('InvalidSequenceTokenException, retrying', true)
+          // InvalidSequenceToken means we need to do a describe to get another token
+          // also do the same if ResourceNotFound as that will result in  the last token for the group being set to null
+          // and ensureGroup + getStream being called which results in the resource being created if not exists
```

Comment:
I would cut the comment to 

```JS
// InvalidSequenceToken means we need to do a describe to get another token
// also do the same if ResourceNotFound as that will result in the last token 
// for the group being set to null
```

What do you think? My reasoning is that the part I've left explains why we are doing the next lines of code, while the part I've removed is duplicating the code, IMHO.